include/turtle/mock.hpp

```diff
@@ -21,133 +21,126 @@
 #include "detail/parameter.hpp"
 #include <boost/preprocessor/repetition/repeat.hpp>
 #include <boost/preprocessor/stringize.hpp>
-#include <boost/utility/identity_type.hpp>
-#include <boost/mpl/assert.hpp>
 
+/// MOCK_CLASS( name )
+/// Define a class
 #define MOCK_CLASS(T) \
     struct T : mock::object
 
-#define MOCK_FUNCTION_TYPE(S, tpn) \
-    tpn boost::remove_pointer< tpn BOOST_IDENTITY_TYPE(S) >::type
+/// MOCK_PROTECT_FUNCTION_SIG( signature )
+/// Use this with MOCK_FUNCTION/MOCK_*_METHOD if the return type contains commas
+#define MOCK_PROTECT_FUNCTION_SIG(...) \
+    mock::detail::parameter_type_t<void (__VA_ARGS__)>
 
-#ifdef MOCK_VARIADIC_MACROS
+/// Internal compatibility macro if function signature is passed via BOOST_IDENTITY_TYPE
+/// TODO: Remove support for doing that and move remove_pointer_t to MOCK_PROTECT_FUNCTION_SIG
+#define MOCK_FUNCTION_TYPE(...) \
+    std::remove_pointer_t< __VA_ARGS__ >
 
+/// MOCK_BASE_CLASS( name, base )
+/// Define a class deriving from a base class
 #define MOCK_BASE_CLASS(T, ...) \
     struct T : __VA_ARGS__, mock::object, mock::detail::base< __VA_ARGS__ >
 
+/// MOCK_FUNCTOR( name, signature )
+/// Define a callable variable/member
 #define MOCK_FUNCTOR(f, ...) \
-    mock::detail::functor< MOCK_FUNCTION_TYPE((__VA_ARGS__),) > f, f##_mock
-#define MOCK_FUNCTOR_TPL(f, ...) \
-    mock::detail::functor< \
-        MOCK_FUNCTION_TYPE((__VA_ARGS__), typename) > f, f##_mock
-
-#else // MOCK_VARIADIC_MACROS
-
-#define MOCK_BASE_CLASS(T, I) \
-    struct T : I, mock::object, mock::detail::base< I >
-
-#define MOCK_FUNCTOR(f, S) \
-    mock::detail::functor< MOCK_FUNCTION_TYPE((S),) > f, f##_mock
-#define MOCK_FUNCTOR_TPL(f, S) \
-    mock::detail::functor< \
-        MOCK_FUNCTION_TYPE((S), typename) > f, f##_mock
-
-#endif // MOCK_VARIADIC_MACROS
+    mock::detail::functor< MOCK_FUNCTION_TYPE(__VA_ARGS__) > f, f##_mock
+/// MOCK_FUNCTOR_TPL( name, signature )
+/// Deprecated. Same as MOCK_FUNCTOR
+#define MOCK_FUNCTOR_TPL(f, ...) MOCK_FUNCTOR(f, __VA_ARGS__)
```

Comment:
Should we not just remove them? I don’t mind breaking users code in a new release if we give them a simple solution to fix it quickly.
Or maybe actually we should keep the deprecated macros but make it trigger a compilation error explaining how to change the code?

Reply:
I can expand it to `static_assert(false, "MOCK_FUNCTOR_TPL has been replaced by MOCK_FUNCTOR")` :)

Reply:
Nice 😁